src/ape/managers/chain.py

```diff
@@ -975,71 +980,100 @@ def get(
             address (AddressType): The address of the contract.
             default (Optional[ContractType]): A default contract when none is found.
               Defaults to ``None``.
+            cache_to_disk (bool): Cache the fetched contract type to  (live networks only).
+              Defaults to ``True``.
+            fetch_from_explorer (bool): Fetch the contract type from the block explorer.
+              Defaults to ``True``.
+            force_disk_cache_update (bool): Force an update to the cache.
+              Defaults to ``False``.
 
         Returns:
             Optional[ContractType]: The contract type if it was able to get one,
               otherwise the default parameter.
         """
-
         address_key: AddressType = self.conversion_manager.convert(address, AddressType)
-        contract_type = self._local_contract_types.get(address_key)
-        if contract_type:
-            if default and default != contract_type:
-                # Replacing contract type
-                self._local_contract_types[address_key] = default
-                return default
-
-            return contract_type
 
+        # Handle the local network case first
         if self._network.name == LOCAL_NETWORK_NAME:
             # Don't check disk-cache or explorer when using local
             if default:
                 self._local_contract_types[address_key] = default
 
             return default
 
-        contract_type = self._get_contract_type_from_disk(address_key)
-        if not contract_type:
+        # Attempt to get the contract type from memory
+        contract_type_from_memory = self._local_contract_types.get(address_key)
+
+        # Attempt to get the contract type from disk
+        contract_type_from_disk = self._get_contract_type_from_disk(address_key)
```

Comment:
Is the primary purpose of the memory cache to avoid disk I/O? I can rework the logic once I better understand the modality of using one or the other.

Reply:
It's simple enough to restore the previous behavior (return the contract type immediately if found in memory). 

Is there a means whereby the memory cache could have some stale state that you'd like to ignore, then continue to the block explorer for a refresh & update? If so, it may be better to ignore or remove the existing entry in the memory cache if some forced update is being performed.

@fubuloubu suggested a kwarg to ignore the cache, which may be more intuitive than interpreting intent based on the set of kwargs I've added thus far.

Reply:
ya, i suppose we need to add kwargs for that too, so we can force update from explorer and what not.